### ISSUE TITLE
Slide with fletcher diagram and subsection generates different heading numbers

### ISSUE THREAD

**Issue by Pamplemousse** [touying-typ/touying#83]
In a slide containing a fletcher diagram, I wanted to add a subsection title.
However, the same subsection will take different heading number accross the slides.

The following source file:
```typst
// mwe.typ
#import "@preview/fletcher:0.5.1" as fletcher: edge, node
#import "@preview/touying:0.5.2": pause, slide, themes, touying-reducer
#import themes.metropolis: *

#show: metropolis-theme
#set heading(numbering: "1.1")

#let fletcher-diagram = touying-reducer.with(reduce: fletcher.diagram, cover: fletcher.hide)

= foo

#slide[
  == bar

  #fletcher-diagram(
    node-stroke: .1em,
    spacing: 3em,

    node((0, 1), `user`),
    pause,
    node((1, 1), `application`),
    edge((0, 1), (1, 1), `uses`, "->"),
  )
]
```

compiled with the following command:
```shell
typst watch --format png mwe.typ mwe{n}.png
```

then we get the following three slides:

![mwe1](https://github.com/user-attachments/assets/3ddf47b5-5fd6-4db9-9f6c-ded020435332)
![mwe2](https://github.com/user-attachments/assets/36889ef6-44d0-43d8-b5df-125720837ec6)
![mwe3](https://github.com/user-attachments/assets/5e1b713e-2824-428c-8531-56c9d084a712)

I expected the heading for `bar` not to change as the rest of the diagram is revealed (the more `pause` are added, the more the heading gets incremented).
Note that this example also highlights #84 which I separated because it doesn't seem related to `fletcher`.

**Comment by OrangeX4** [comment]
Duplicated with #38. We need to wait for frozen state to do this.

**Comment by OrangeX4** [comment]
And you can also consider:

```typst
#import "@preview/fletcher:0.5.1" as fletcher: edge, node
#import "@preview/touying:0.5.2": pause, slide, themes, touying-reducer
#import themes.metropolis: *

#show: metropolis-theme.with(
  header: utils.display-current-heading(level: 1),
  config-common(
    subslide-preamble: text(weight: "bold", utils.display-current-heading(level: 2)) + parbreak(),
  )
)
#set heading(numbering: "1.1")

#let fletcher-diagram = touying-reducer.with(reduce: fletcher.diagram, cover: fletcher.hide)

= foo

== bar

#fletcher-diagram(
  node-stroke: .1em,
  spacing: 3em,

  node((0, 1), `user`),
  pause,
  node((1, 1), `application`),
  edge((0, 1), (1, 1), `uses`, "->"),
)
```

![image](https://github.com/user-attachments/assets/8572a00c-d411-40c2-b931-b81624d33ec6)

I think it's a more elegant method.

**Comment by Pamplemousse** [comment]
> I think it's a more elegant method.

But it breaks anything more complicated than the minimum example:
* Slides that have content (like a little introduction) before a subsection. You get
```
== heading
little intro
== heading
content
```
instead of the expected
```
little intro
== heading
content
```
* It doesn't handle multiple levels throughout the presentation, for example a slide with a diagram under a subsection, and another slide with a diagram under a subsubsection.

**Comment by OrangeX4** [comment]
Fixed at #124. Wait for the next release.
